src/core/zowe/core_for_zowe_sdk/profile_manager.py

```diff
@@ -259,10 +260,17 @@ def load(
 
         missing_secure_props = []  # track which secure props were not loaded
 
+        loaded_cfg: dict = {}
+
         for i, (config_type, cfg) in enumerate(config_layers.items()):
+            loaded_cfg = always_merger.merge(loaded_cfg, cfg.profiles)
+            if loaded_cfg:
+                cfg.profiles = loaded_cfg
+                
             profile_loaded = self.get_profile(
```

Comment:
This `get_profile` call and the lines that follow it could be moved outside of the for loop. We only need to load profile properties, secure properties, and environment variables once after we have the merged config object.